src/prefect/settings/legacy.py

```diff
@@ -80,11 +81,38 @@ def _env_var_to_accessor(env_var: str) -> str:
     """
     Convert an environment variable name to a settings accessor.
     """
-    if SETTING_VARIABLES.get(env_var) is not None:
-        return SETTING_VARIABLES[env_var].accessor
+    if (field := _get_settings_fields(Settings).get(env_var)) is not None:
+        return field.accessor
     return env_var.replace("PREFECT_", "").lower()
 
 
+@cache
+def _get_valid_setting_names(cls: type[BaseSettings]) -> Set[str]:
```

Comment:
or is this fine in 3.9?
```suggestion
def _get_valid_setting_names(cls: Type[BaseSettings]) -> Set[str]:
```